tools/windows/install-help/cal/CustomAction.cpp

```diff
@@ -5,24 +5,27 @@ extern "C" UINT __stdcall FinalizeInstall(MSIHANDLE hInstall)
 
     HRESULT hr = S_OK;
     UINT er = ERROR_SUCCESS;
-    CustomActionData data;
-    // first, get the necessary initialization data
-    // need the dd-agent-username (if provided)
-    // need the dd-agent-password (if provided)
+    std::optional<CustomActionData> data;
     hr = WcaInitialize(hInstall, "CA: FinalizeInstall");
     ExitOnFailure(hr, "Failed to initialize");
     WcaLog(LOGMSG_STANDARD, "Initialized.");
 
 #ifdef _DEBUG
     MessageBox(NULL, L"hi", L"bye", MB_OK);
 #endif
-    if (!data.init(hInstall))
-    {
+    // first, get the necessary initialization data
+    // need the dd-agent-username (if provided)
+    // need the dd-agent-password (if provided)
+    try {
+        auto propertyView = std::make_shared<DeferredCAPropertyView>(hInstall);
```

Comment:
Is it worth it to change the data type. It trigger lot of signature changes all over the place (in number functions it its the only change)